activemodel/lib/active_model/dirty.rb

```diff
@@ -130,6 +131,22 @@ module Dirty
       attribute_method_affix prefix: "restore_", suffix: "!"
     end
 
+    def initialize_dup(other) # :nodoc:
+      super
+      @attributes = self.class._default_attributes.map do |attr|
+        attr.with_value_from_user(@attributes.fetch_value(attr.name))
+      end
+      @mutations_from_database = nil
+    end
+
+    def changes_applied # :nodoc:
+      @previously_changed = changes
```

Comment:
Can we eliminate this ivar entirely now and just use the computed changes?

Reply:
It's needed for Dirty to continue to support old ivar backed attributes.

Reply:
Don't ivar backed attributes always need to call `attribute_will_change!` manually? Is it possible for us to somehow work that into `AttributeMutationTracker.force_changes`?

Reply:
The ivar ones aren't in there.  I could investigate putting them in there, but my impression was that we're deprecating them, so I didn't want to spend too much time refactoring how they're tracked.  What my code does right now very much mirrors for AM how it's been dealt with in AR through the use of `super` calls in `AR::Dirty`.